code/game/machinery/bots/draculabot.dm

```diff
@@ -0,0 +1,278 @@
+#define DEFAULT_DRINK_RATE 8 //2x IV drip speed, finishes with a patient in ~7 seconds
+#define DANGER_DRINK_RATE 40 //Terrifying speed, can drain a human in 14 seconds or begin to cause damage in 6 seconds
+//Notes: Humans have 560u Blood. This bot won't consume below 501u (~90%). Therefore, 59u can be donated if full.
+//Humans regain 0.1u blood/tick, +0.6u/t for 0.5 nutriment, +1.2u/t for 0.5 iron.
+//One ijzerkoekje is theoretically good for 18u blood over 10 ticks.
+//Rewards dispensed once every 55u donated.
+
+/obj/machinery/bot/bloodbot
+	name = "Doctor Acula"
+	desc = "A blood donation medibot. If wearing a cape, notify security immediately."
+	icon = 'icons/obj/aibots.dmi'
+	icon_state = "bloodbot00"
+	density = 0
+	on = 0
+	anchored = 0
+	health = 40
+	maxhealth = 20
+	req_access = list(access_medical)
+	var/mob/living/carbon/human/target //Only used if emagged
+	var/currently_drawing_blood = 0 //One patient at a time.
+	var/quiet = 0
+	var/since_last_reward = 0 //Once every 55u, dispense reward
+	var/list/possible_rewards = list(/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje,
+								/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje,
+								/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje,
+								/obj/item/weapon/reagent_containers/food/snacks/chococoin/wrapped,
+								/obj/item/weapon/reagent_containers/food/snacks/chococoin/wrapped,
+								/obj/item/weapon/reagent_containers/food/snacks/chococoin/wrapped,
+								/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje_helper_dummy) //3/7 chance for one cookie, 3/7 chance for chococoin, 1/7 for many cookies!
+	var/list/contained_bags = list()
+	var/obj/item/weapon/reagent_containers/blood/last_bag
+
+/obj/machinery/bot/bloodbot/New()
+	..()
+	for(var/i = 1 to 8)
+		contained_bags += new /obj/item/weapon/reagent_containers/blood/empty(src)
+
+/obj/machinery/bot/bloodbot/Destroy()
+	for(var/obj/O in contained_bags)
+		O.forceMove(get_turf(src))
+	contained_bags = null
+	last_bag = null
+	..()
+
+/obj/machinery/bot/bloodbot/update_icon()
+	if(contained_bags.len || emagged)
+		icon_state = "bloodbot[emagged][currently_drawing_blood]"
+	else
+		icon_state = "bloodbot-e"
+
+/obj/machinery/bot/bloodbot/turn_on()
+	..()
+	update_icon()
+	updateUsrDialog()
+
+/obj/machinery/bot/bloodbot/turn_off()
+	if(emagged)
+		return
+	..()
+	currently_drawing_blood = 0
+	update_icon()
+
+/obj/machinery/bot/bloodbot/attack_paw(mob/user as mob)
+	return attack_hand(user)
+
+/obj/machinery/bot/bloodbot/attack_hand(mob/user as mob)
+	if(..())
+		return
+	var/dat
+	dat += "<TT><B>Acula-class Blood Donation Bot v1.0</B></TT><BR><BR>"
```

Comment:
This makes the font impact or whatever, it's a sarif font

I use it for the bloodbags down below too, just to make the text look different